src/twisted/conch/ssh/transport.py

```diff
@@ -1275,31 +1279,44 @@ def _ssh_KEX_ECDH_INIT(self, packet):
         pubHostKey = self.factory.publicKeys[self.keyAlg]
         privHostKey = self.factory.privateKeys[self.keyAlg]
 
-        # Get the curve instance
-        try:
-            curve = keys._curveTable[b'ecdsa' + self.kexAlg[4:]]
-        except KeyError:
-            raise UnsupportedAlgorithm('unused-key')
-
-        # Generate the private key
-        ecPriv = ec.generate_private_key(curve, default_backend())
-
-        # Get the public key
-        ecPub = ecPriv.public_key()
-        encPub = ecPub.public_numbers().encode_point()
-
-        # Take the provided public key and transform it into
-        # a format for the cryptography module
-        theirECPub = ec.EllipticCurvePublicNumbers.from_encoded_point(
-                        curve, pktPub).public_key(default_backend())
-
-        # We need to convert to hex,
-        # so we can convert to an int
-        # so we can make it a multiple precision int.
-        sharedSecret = MP(
-                       int(
-                        binascii.hexlify(
-                          ecPriv.exchange(ec.ECDH(), theirECPub)), 16))
+        if self.kexAlg.find(b'25519') != -1:
+            # Turn the public key into a public key object
+            theirECPub = nacl.public.PublicKey(pktPub)
+
+            # Always generate new private and public keys
+            ecPriv = nacl.public.PrivateKey.generate()
+            ecPub = ecPriv.public_key
+            encPub = ecPub._public_key
```

Comment:
Yup.  If you use the standard method it will return a value that works for pynacl, but won't work for almost every other implementation, including OpenSSH.

Why they do this I have no idea, I'm sure they have their reasons, but this was the only way I could get it to work.